### DISCUSSION TITLE
How can I change the default 'admin' username to another one?

### DISCUSSION THREAD

**Discussion by stripedgoat** [owncast/owncast#1740] [Q&A] [answered]
I'd like to change the default 'admin' username to something else. An additionally welcome thing would be having a different password from the streamkey.

Is there a way to change these credentials?

It would be necessary for me because I'd hand out the streamkey to friends, so they can stream on my server... But the problem is, if I give them the streamkey, they can log in to the admin area with just a few guesses.

Thanks in advance for the answer!

**Comment by gabek** [comment, marked as answer]
There is not currently any way to change the username or having different passwords from the stream key.

**Reply by stripedgoat** [reply]
Got it. Thanks! Perhaps a way to change the /admin url ending to another one? (access to admin panel)

**Reply by aaronpk** [reply]
You could configure the `/admin` URL to be accessible only by certain IPs in Apache or nginx. Something like this:

nginx:

```nginx
location /admin {
    allow 1.2.3.4;
    deny all;
    ...
}
```

apache:

```apache
<Location /admin>
Order allow,deny
Allow from 192.168.1.0/24
...
</Location>
```

**Reply by stripedgoat** [reply]
That'll do! Thank you. 🙏🏼

**Reply by sitehatchery** [reply]
This is a featured I'd also like added. It's too easy to figure out how to get into admin and change settings, if the stream key is known.

**Reply by gabek** [reply]
The stream key should be kept private and treated as a password.

**Reply by sitehatchery** [reply]
Can't assume that only one person will use it. If multiple people can stream, then you have to give each of them the streaming key. And... any one of those people can easily figure out how to get into admin.

**Reply by gabek** [reply]
> Can't assume that only one person will use it.

Owncast is single-tenant software that is absolutely assuming one person is using it. While you can do with it what you'd like, and give your key out to as many people as you want, that's a choice you are personally making. With Owncast the streamer is the admin. If you want more people to stream on one server then they are all the admin.

A streamer that isn't an admin means they can't moderate chat, they can't look at viewer details, they can't troubleshoot issues, they can't change the title of the stream, etc.  I understand this may not matter to you specifically, but this is all important stuff.

**Reply by sitehatchery** [reply]
> If you want more people to stream on one server then they are all the admin.

They don't need to access admin. They just need to put the streaming URL and stream key into their streaming software (OBS in my case). 

> A streamer that isn't an admin means they can't moderate chat, they can't look at viewer details, they can't troubleshoot issues, they can't change the title of the stream, etc. I understand this may not matter to you specifically, but this is all important stuff.

Yes, that's true. They can't do these things if all they have is a streaming URL and stream key, unless they determine how to log into admin. The moderator/designated admin will do this. But in our case, the streamer/broadcaster will not have access to this.

All that's really needed to make it more secure is to make the "admin" user anything I want from command line.

**Comment by DidierMalenfant** [comment]
+1 on this to allow streamers that are not admin
